lib/ical/helpers.js

```diff
@@ -1,377 +1,314 @@
+ * @return {String}               The number padded as a string
+ */
+export function pad2(data) {
+  if (typeof(data) !== 'string') {
+    // handle fractions.
+    if (typeof(data) === 'number') {
+      data = parseInt(data);
     }
-  },
+    data = String(data);
+  }
 
-  /**
-   * Truncates the given number, correctly handling negative numbers.
-   *
-   * @param {Number} number     The number to truncate
-   * @return {Number}           The truncated number
-   */
-  trunc: function trunc(number) {
-    return (number < 0 ? Math.ceil(number) : Math.floor(number));
-  },
+  let len = data.length;
 
-  /**
-   * Poor-man's cross-browser inheritance for JavaScript. Doesn't support all
-   * the features, but enough for our usage.
-   *
-   * @param {Function} base     The base class constructor function.
-   * @param {Function} child    The child class constructor function.
-   * @param {Object} extra      Extends the prototype with extra properties
-   *                              and methods
-   */
-  inherits: function(base, child, extra) {
-    function F() {}
-    F.prototype = base.prototype;
-    child.prototype = new F();
+  switch (len) {
+    case 0:
+      return '00';
+    case 1:
+      return '0' + data;
+    default:
+      return data;
+  }
+}
 
-    if (extra) {
-      ICAL.helpers.extend(extra, child.prototype);
-    }
-  },
+/**
+ * Truncates the given number, correctly handling negative numbers.
+ *
+ * @param {Number} number     The number to truncate
+ * @return {Number}           The truncated number
+ */
+export function trunc(number) {
+  return (number < 0 ? Math.ceil(number) : Math.floor(number));
+}
 
-  /**
-   * Poor-man's cross-browser object extension. Doesn't support all the
-   * features, but enough for our usage. Note that the target's properties are
-   * not overwritten with the source properties.
-   *
-   * @example
-   * var child = ICAL.helpers.extend(parent, {
-   *   "bar": 123
-   * });
-   *
-   * @param {Object} source     The object to extend
-   * @param {Object} target     The object to extend with
-   * @return {Object}           Returns the target.
-   */
-  extend: function(source, target) {
-    for (var key in source) {
-      var descr = Object.getOwnPropertyDescriptor(source, key);
-      if (descr && !Object.getOwnPropertyDescriptor(target, key)) {
-        Object.defineProperty(target, key, descr);
-      }
+/**
+ * Poor-man's cross-browser object extension. Doesn't support all the
+ * features, but enough for our usage. Note that the target's properties are
+ * not overwritten with the source properties.
+ *
+ * @example
+ * var child = ICAL.helpers.extend(parent, {
+ *   "bar": 123
+ * });
+ *
+ * @param {Object} source     The object to extend
+ * @param {Object} target     The object to extend with
+ * @return {Object}           Returns the target.
+ */
+export function extend(source, target) {
+  for (let key in source) {
```

Comment:
How is this function `extend` better than `Object.assign()`?  Is ical.js supposed to work in ancient environments with defect `Object.assign()`?

Reply:
Alternatively:

```js
let  newObj = { ...oldObj }
```

Reply:
Certainly something we can remove. This was from times before Object.assign. The PR is already large though, lets take care in a follow up.